freqtrade/data/dataprovider.py

```diff
@@ -105,6 +105,17 @@ def historic_ohlcv(
             )
         return self.__cached_pairs_backtesting[saved_pair].copy()
 
+    def get_required_startup_seconds(self, timeframe: str) -> int:
+        tf_seconds = timeframe_to_seconds(timeframe)
+        base_seconds = tf_seconds * self._config.get('startup_candle_count', 0)
+        if not self._config['freqai']['enabled']:
+            return base_seconds
+        else:
+            train_seconds = self._config['freqai']['train_period_days'] * 86400
+            # multiplied by safety factor of 2 because FreqAI users
+            # typically do not know the correct window.
+            return base_seconds * 2 + int(train_seconds)
```

Comment:
i wonder if this could be combined with `get_required_startup()` from backtesting.
They both use the same logic - with the only difference being that this returns seconds, while this returns candles - so maybe moving the function from backtesting here, and calling that could reduce code duplication?

Reply:
I combined them and removed the redundant code from both `backtesting.py` and `data_kitchen.py` :-)